### PR TITLE
Plans: Show a Confirmation Dialog When Upgrading Will Launch Site

### PR DESCRIPTION
**Pull request by jblz**
Note: This is targeting the branch in #35003. Let's see if this is a better way to handle this & merge it in there if so.

#### Changes proposed in this Pull Request

* Instead of showing a notice in the plan description, show an "are you sure" dialog

#### Testing instructions

* Follow test plan in #35003, but instead of seeing a notice on the plan, you should be asked to confirm your action before you're taken to the launch flow.

### CODE REVIEW

**Comment by jblz** [comment]
Right now, this is pretty simple:

Screen Shot 2019-08-07 at 12 06 13 PM

If we decide to go this route, we can add to the design as we see fit.

File: client/my-sites/plan-features/index.jsx [hunk 1]
```diff
@@ -481,6 +563,53 @@ export class PlanFeatures extends Component {
...
+
+		const {
+			availableForPurchase,
+			cartItemForPlan,
+			isPrivateAndGoingAtomic,
+			planPath,
+			productSlug,
+		} = singlePlanProperties;
+
+		if ( ownPropsOnUpgradeClick && ownPropsOnUpgradeClick !== noop && cartItemForPlan ) {
```
Comment (spen) [inline client/my-sites/plan-features/index.jsx]: if it were a `noop`, wouldn't it be fine to allow it to be called?
Would `isFunction( ownPropsOnUpgradeClick ) && ...` suffice here?

File: client/my-sites/plan-features/index.jsx [hunk 2]
```diff
@@ -481,6 +563,53 @@ export class PlanFeatures extends Component {
...
+			return;
+		}
+
+		const args = {};
+		// Auto-apply the coupon code to the cart for WPCOM sites
+		if ( ! displayJetpackPlans && withDiscount ) {
+			args.coupon = withDiscount;
+		}
+
+		if ( isPrivateAndGoingAtomic ) {
```
Comment (spen) [inline client/my-sites/plan-features/index.jsx]: What do you think to moving this conditional up above `const args = ...` to keep the args logic together with its usage in `page( addQuery`?

File: client/my-sites/plan-features/index.jsx [hunk 3]
```diff
@@ -869,45 +974,12 @@ export default connect(
 					isLandingPage,
 					isPlaceholder,
 					isPrivateAndGoingAtomic,
-					onUpgradeClick: onUpgradeClick
```
Comment (spen) [inline client/my-sites/plan-features/index.jsx]: Good shout btw!

File: client/my-sites/plan-features/index.jsx (hunk 1 above)
Comment (jblz) [reply client/my-sites/plan-features/index.jsx]: Ya, the issue is that it defaults to `noop` and this change does not override the prop via `connect` as it did before.

**Comment by monikaburman** [comment]
Sorry for the delay; not a blocker to what is proposed, and fine to go out as-is, but the wording is awkward to me. Suggesting:

Upgrading your plan makes your site visible to the public.

**Comment by jblz** [comment]
Thanks, @monikaburman -- I'll incorporate the copy change in #35003.

**Pull request merged**